src/stores/useLabyrinthStore.jsx

```diff
@@ -0,0 +1,77 @@
+import { create } from "zustand";
+
+export const useLabyrinthStore = create((set, get) => ({
+  loading: false,
+  gameFlow: false,
+  start: {},
+  userName: "",
+  actions: [],
+  description: "",
+  coordinates: "",
+
+  setUserName: (userInput) => set({ userName: userInput }),
+
+  fetchStart: async (userName) => {
+    set({ loading: true });
+    set({ gameFlow: false });
+
+    try {
+      const response = await fetch("https://labyrinth.technigo.io/start", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          username: userName,
```

Comment:
Here you could make the username unique in some way to let the user type in what they want. You could add a random number after the username, or the current date as a string, for example.

        const uniqueUsername = username + Date.now();
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({
          username: uniqueUsername,

/Martin & Etna

Reply:
Great idea!